homeassistant/components/portainer/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Portainer integration."""
+
+from __future__ import annotations
+
+import logging
+
+from pyportainer import (
+    Portainer,
+    PortainerAuthenticationError,
+    PortainerConnectionError,
+    PortainerTimeoutError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryError,
+    ConfigEntryNotReady,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .coordinator import PortainerCoordinator
+from .models import PortainerData
+
+_LOGGER = logging.getLogger(__name__)
+
+_PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR]
+
+type PortainerConfigEntry = ConfigEntry[PortainerData]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
+    """Set up Portainer from a config entry."""
+    if CONF_API_KEY not in entry.data:
+        raise ConfigEntryAuthFailed
+
+    _LOGGER.debug("Setting up Portainer API: %s", entry.data[CONF_HOST])
+
+    api_url = f"{entry.data[CONF_HOST]}:{entry.data[CONF_PORT]}"
+
+    client = Portainer(
+        api_url=api_url,
+        api_key=entry.data[CONF_API_KEY],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        endpoints = await client.get_endpoints()
+    except PortainerAuthenticationError as err:
+        raise ConfigEntryError(
+            f"Invalid Portainer authentication. Error: {err}"
+        ) from err
```

Comment:
you should use exception translations
```suggestion
        raise ConfigEntryError(
            translation_domain=DOMAIN,
            translation_key="invalid_auth",
            translation_placeholders={"error": err}
        ) from err
```